src/HotChocolate/Core/src/Types.Scalars/UtcOffsetType.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.Collections.Generic;
+using HotChocolate.Language;
+
+namespace HotChocolate.Types
+{
+    /// <summary>
+    /// The `UtcOffset` scalar type represents a value of format ±hh:mm.
+    /// </summary>
+    public class UtcOffsetType : ScalarType<TimeSpan, StringValueNode>
+    {
+        private static readonly Dictionary<TimeSpan, string> _timeSpanToOffset =
```

Comment:
You may want to create a nested class to hold this things. It's just nice to work on the dictionaries when you have a constructor to do stuff :) 
 
At the end of the class:

```csharp
private class OffsetLookup 
{
     private readonly Dictionary<TimeSpan, string> _timeSpanToOffset;
     ....
     public OffsetLookup() 
     {
          //...fill the properties  
     }
     
     public bool TrySerialize(TimeSpan value, [NotNullWhen(true)] out string? result) {..}
     public bool TryDeserialize(string value, [NotNullWhen(true)] out TimeSpan? result) {..}
    
     ....
     public static readonly OffsetLookup = new OffsetLookup(); 
}
```

With this you can then just do 

```csharp
 public override IValueNode ParseResult(object? resultValue)
 {
     return resultValue switch
     {
          null => NullValueNode.Default,
          string s when OffsetLookup.TryParse(s, out TimeSpan? timespan) => ParseValue(timespan),
          TimeSpan ts => ParseValue(ts),
          _ => throw ThrowHelper.UtcOffset_ParseValue_IsInvalid(this)
      };
}
```